hi-ml/src/health_ml/utils/lightning_loggers.py

```diff
@@ -0,0 +1,163 @@
+                else:
+                    current_results[key] = value
+        else:
+            self.results_per_epoch[epoch] = metrics  # type: ignore
+
+    @rank_zero_only
+    def log_hyperparams(self, params: Any) -> None:
+        self.hyperparams = params
+
+    def experiment(self) -> Any:
+        return None
+
+    def name(self) -> Any:
+        return ""
+
+    def version(self) -> int:
+        return 0
+
+    @property
+    def epochs(self) -> Iterable[int]:
+        """
+        Gets the epochs for which the present object holds any results.
+        """
+        return self.results_per_epoch.keys()
+
+    def extract_by_prefix(self, epoch: int, prefix_filter: str = "") -> DictStrFloat:
+        """
+        Reads the set of metrics for a given epoch, filters them to retain only those that have the given prefix,
+        and returns the filtered ones. This is used to break a set
+        of results down into those for training data (prefix "Train/") or validation data (prefix "Val/").
+
+        :param epoch: The epoch for which results should be read.
+        :param prefix_filter: If empty string, return all metrics. If not empty, return only those metrics that
+        have a name starting with `prefix`, and strip off the prefix.
+        :return: A metrics dictionary.
+        """
+        epoch_results = self.results_per_epoch.get(epoch, None)
+        if epoch_results is None:
+            raise KeyError(f"No results are stored for epoch {epoch}")
+        filtered = {}
+        for key, value in epoch_results.items():
+            assert isinstance(key, str), f"All dictionary keys should be strings, but got: {type(key)}"
+            # Add the metric if either there is no prefix filter (prefix does not matter), or if the prefix
+            # filter is supplied and really matches the metric name
+            if (not prefix_filter) or key.startswith(prefix_filter):
+                stripped_key = key[len(prefix_filter):]
+                filtered[stripped_key] = value  # type: ignore
+        return filtered  # type: ignore
+
+    def to_metrics_dicts(self, prefix_filter: str = "") -> Dict[int, DictStrFloat]:
+        """
+        Converts the results stored in the present object into a two-level dictionary, mapping from epoch number to
+        metric name to metric value. Only metrics where the name starts with the given prefix are retained, and the
+        prefix is stripped off in the result.
+
+        :param prefix_filter: If empty string, return all metrics. If not empty, return only those metrics that
+        have a name starting with `prefix`, and strip off the prefix.
+        :return: A dictionary mapping from epoch number to metric name to metric value.
+        """
+        return {epoch: self.extract_by_prefix(epoch, prefix_filter) for epoch in self.epochs}
+
+    def get_metric(self, is_training: bool, metric_type: str) -> List[float]:
```

Comment:
This function is probably specific to InnerEye, where we use train/ and val/ as prefixes. Retire?